### ISSUE TITLE
Title in kitty is always `__MISE_DIR`

### ISSUE THREAD

**Issue by rmehner** [jdx/mise#3345]
**Describe the bug**
Since a couple of versions, tab titles in kitty always show as `~__MISE_DIR`, instead of the actual title (current work dir or process). This only happens for working directories with a mise config (no matter of legacy version files or actual `.mise.toml`)

**To Reproduce**
1. Use kitty (0.37.0)
2. Use mise (2024.11.37 macos-arm64 (2024-11-30))
3. Go to a directory with a version file
4. See `~__MISE_DIR` as tab title

![kitty-2024-12-04-11-14-03@2x](https://github.com/user-attachments/assets/390ce12f-a683-4c18-9b02-f612bcb2c0da)

**Expected behavior**
See the actual current working directory / process, as it is the default for kitty.

**`mise doctor` output**

```text
version: 2024.11.37 macos-arm64 (2024-11-30)
activated: yes
shims_on_path: no

build_info:
  Target: aarch64-apple-darwin
  Features: DEFAULT, NATIVE_TLS
  Built: Sat, 30 Nov 2024 00:54:51 +0000
  Rust Version: rustc 1.82.0 (f6e511eec 2024-10-15) (Homebrew)
  Profile: release

shell:
  /opt/homebrew/bin/zsh
  zsh 5.9 (arm-apple-darwin22.1.0)

dirs:
  cache: ~/Library/Caches/mise
  config: ~/.config/mise
  data: ~/.local/share/mise
  shims: ~/.local/share/mise/shims
  state: ~/.local/state/mise

config_files:
  ~/.tool-versions
  ~/.config/mise/config.toml
  ~/work/projects/project-path/.ruby-version
  ~/work/projects/project-path/Gemfile
  ~/work/projects/project-path/.mise.toml

backends:
  aqua
  asdf
  cargo
  core
  gem
  go
  npm
  pipx
  spm
  ubi
  vfox

plugins:

toolset:
  core:node@20.18.0
  core:python@3.12.3
  core:ruby@3.3.6

env_vars:
  MISE_SHELL=zsh

settings:

No problems found
```

**Additional context**
I realize this might also be a kitty bug, as it doesn't happen with Terminal.app. It didn't happen with older versions of mise though, so I thought I report here first :) I'm happy about any pointers. Thank you for your excellent work, enjoying mise very much for quite a while already.

**Comment by jdx** [comment]
I use kitty and don't see this behavior. I tried with zsh.

**Comment by rmehner** [comment]
@jdx interesting. What's your `tab_title_template` config setting looking like? I've commented mine out to test if my config setting is the culprit.

My current one is set like this: `tab_title_template "{index}{fmt.fg.red}{bell_symbol}{activity_symbol}{fmt.fg.tab} {title}"`

**Comment by jdx** [comment]
```
~ ❯ cat ~/.config/kitty/*.conf | grep tab_title_template
#: tab_title_template for how this is rendered.
#: tab_title_template for how this is rendered.
# tab_title_template "{fmt.fg.red}{bell_symbol}{activity_symbol}{fmt.fg.tab}{title}"
# active_tab_title_template none
#: tab_title_template.
tab_title_template          {title}{' :{}:'.format(num_windows) if num_windows > 1 else ''}
```

**Comment by jdx** [comment]
that `__MISE_DIR` was recently added as an env var in `hook-env` but I have no idea why kitty would be exposing it. It's just one of 3 env vars that mise exports for its state.

**Comment by rmehner** [comment]
I don't even have `~/.zshenv`. 

I have `eval "$(mise activate zsh)"` in my `~/.profile` which is evaluated by `~/.zshrc`.

Btw. thank you so much for looking into this!

**Comment by jdx** [comment]
using `~/.profile` could be the issue

**Comment by ShikChen** [comment]
I had `eval "$(mise activate zsh)"` in my `~/.zshrc` and encountered a similar issue with my prompt. As a workaround, I switched to a shim-based solution by adding `~/.local/share/mise/shims` to my `$PATH`. For more details on shims, see [Mise documentation](https://mise.jdx.dev/dev-tools/shims.html).

**Comment by rmehner** [comment]
Found the issue!

My `~/.zshrc` has the following line:

```sh
setopt AUTO_NAME_DIRS
```

With this enabled, the problem will appear. If I remove that line, it works just fine, just the dir names in the prompt/tab titles are very long now, since it cannot use this feature anymore.

**Comment by rmehner** [comment]
And another thing:

If I disable the [shell_integration](https://sw.kovidgoyal.net/kitty/shell-integration/) in kitty, it also works with `AUTO_NAME_DIRS` being enabled.

Also setting it to `no-rc` or `no-title` works.

**Comment by ShikChen** [comment]
I do use `auto_name_dirs` intentionally so it would be great if mise can avoid the conflict in env vars.

**Comment by jdx** [comment]
I'll probably do something like prefix this var with a ":". It's really dumb I can't just set an env var to a directory name.

**Comment by jdx** [comment]
can one of you test this to see if it resolves your issue https://github.com/jdx/mise/pull/3366

**Comment by ShikChen** [comment]
Yes, it resolves the `__MISE_DIR` zsh issue. Thanks for the quick fix!
